lib/ansible/modules/cloud/amazon/aws_glacier_vault.py

```diff
@@ -0,0 +1,227 @@
+
+DOCUMENTATION = '''
+---
+module: aws_glacier_vault
+short_description: Manage Glacier vaults in AWS
+description:
+    - Manage Glacier vaults in AWS
+version_added: "2.6"
+requirements: [ boto3 ]
+author: "Loic Blot <loic.blot@unix-experience.fr>"
+options:
+  name:
+    description:
+      - Name of the glacier vault
+    required: true
+    default: null
+  state:
+    description:
+      - Create or remove the glacier vault
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  tags:
+    description:
+      - A hash/dictionary of tags to add to the new instance or to add/remove from an existing one.
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+RETURN = '''
+vault:
+    description: the vault object
+    returned: always
+    type: complex
+    contains:
+        CreationDate:
+            description: The vault creation date.
+            returned: always
+            type: string
+            sample: "2018-02-13T22:10:28.590Z"
+        NumberOfArchives:
+            description: Number of archives currently in the vault.
+            returned: always
+            type: int
+            sample: 0
+        SizeInBytes:
+            description: Current vault size.
+            returned: always
+            type: int
+            sample: 0
+        VaultARN:
+            description: The vault AWS ARN.
+            returned: always
+            type: string
+            sample: "arn:aws:glacier:eu-west-3:123457788994:vaults/test-vault"
+        VaultName:
+            description: The vault name.
+            returned: always
+            type: string
+            sample: "test-vault"
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create a simple glacier vault
+- aws_glacier_vault:
+    name: myglaciervault
+
+# Create a simple glacier vault in eu-west-3 region
+- aws_glacier_vault:
+    name: myglaciervault
+    region: eu-west-3
+
+# Remove an glacier vault
+- aws_glacier_vault:
+    name: myglaciervault
+    state: absent
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (ec2_argument_spec, boto3_conn, HAS_BOTO3, get_aws_connection_info,
+                                      compare_aws_tags)
+
+try:
+    import botocore
+    from botocore.exceptions import ClientError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+class GlacierVaultManager(object):
+    def __init__(self, connection, module):
+        self.connection = connection
+        self.module = module
+        self.name = self.module.params.get('name')
+
+    def list(self):
+        vaults = self.connection.list_vaults()
+        if "VaultList" not in vaults:
+            self.module.fail_json(msg="Invalid response from Glacier: no VaultList found in response")
+
+        return vaults["VaultList"]
+
+    def exists(self, name):
+        for v in self.list():
```

Comment:
`return name in [v['VaultName'] for v in self.list()]`

will almost certainly work in all cases (if not, use `get` to avoid a KeyError or catch the KeyError)

Reply:
i'm returning the vault object here

Reply:
maybe i should rename it to get_vault instead